CI: ci-awscni & ci-eks - AlreadyExistsException: Stack [eksctl-cilium-cilium-<runid>-1-cluster] already exists

## CI failure

Creating the AWS EKS cluster during the tests `ci-awscni` & `ci-eks` fails with the following failiure:

```
2023-11-28 00:49:50 [ℹ]  building cluster stack "eksctl-cilium-cilium-7012612170-1-cluster"
Error: failed to create cluster "cilium-cilium-7012612170-1"
2023-11-28 00:49:51 [!]  1 error(s) occurred and cluster hasn't been created properly, you may wish to check CloudFormation console
2023-11-28 00:49:51 [ℹ]  to cleanup resources, run 'eksctl delete cluster --region=eu-central-1 --name=cilium-cilium-7012612170-1'
2023-11-28 00:49:51 [✖]  creating CloudFormation stack "eksctl-cilium-cilium-7012612170-1-cluster": operation error CloudFormation: CreateStack, https response error StatusCode: 400, RequestID: 02c84e8b-0c98-449a-8cd6-7c5dcdfadfe9, AlreadyExistsException: Stack [eksctl-cilium-cilium-7012612170-1-cluster] already exists
Error: Process completed with exit code 1.
```

e.g. https://github.com/cilium/cilium/actions/runs/7012612170/job/19077439535

The issue is that the name with the pattern `${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}-${{ github.run_attempt }}` isn't unique enough in cases where a testrun deploys a cluster in the same region twice. This is the case for `eu-central-1` since #29004 

https://github.com/cilium/cilium/blob/main/.github/workflows/conformance-aws-cni.yaml#L54

## Proposal

a) use a different region -> fixed with https://github.com/cilium/cilium/pull/29427
b) use a unique identifier for the name of the cluster to avoid name clashes

--- a/.github/actions/aws/k8s-versions.yaml
+++ b/.github/actions/aws/k8s-versions.yaml
@@ -4,7 +4,7 @@ include:
   - version: "1.23"
     region: eu-central-1
   - version: "1.24"
-    region: eu-central-1
+    region: eu-west-1
   - version: "1.25"
     region: ap-northeast-1
   - version: "1.26"

